File: android/app/src/main/java/com/rightpoint/lite_chat/channel/ChannelConversation.java
package com.rightpoint.lite_chat.channel;

import androidx.annotation.NonNull;

import com.rightpoint.lite_chat.IM.IResolveConversation;
import com.rightpoint.lite_chat.IM.Msg;
import com.rightpoint.lite_chat.IM.huanxin.HXResolveConversation;
import com.rightpoint.lite_chat.MainActivity;

import java.util.List;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;

/**
 * Description：
 * @author Wonder Wei
 * Create date：2020/7/23 4:00 PM 
 */
public class ChannelConversation {
    static final String CHANNEL_CONVERSATION = "com.rightpoint.litechat/conversation";

    public static void connect(FlutterEngine flutterEngine, MainActivity activity) {
        MethodChannel methodChannel = new MethodChannel(flutterEngine.getDartExecutor(),
                CHANNEL_CONVERSATION);

        IResolveConversation iResolveConversation = new HXResolveConversation();

        methodChannel.setMethodCallHandler(new MethodChannel.MethodCallHandler() {
            @Override
            public void onMethodCall(@NonNull MethodCall call,
                                     @NonNull MethodChannel.Result result) {
                if ("getConversations".equals(call.method)) {
                    List<Msg> conversations = iResolveConversation.getConversation();

                    activity.runOnUiThread(new Runnable() {
                        @Override
                        public void run() {
                            result.success(conversations);
                        }
                    });
                } else if ("getMsgList".equals(call.method)) {
                    String username = call.argument("username");
                    List<Msg> msgList = iResolveConversation.getMsgList(username);
                    activity.runOnUiThread(new Runnable() {
                        @Override
                        public void run() {
                            result.success(msgList);
                        }
                    });
                }
            }
        });
    }
}

File: android/app/src/main/java/com/rightpoint/lite_chat/channel/ChannelConversation.kt
package com.rightpoint.lite_chat.channel

import com.hyphenate.chat.EMClient
import com.rightpoint.lite_chat.IM.IResolveConversation
import com.rightpoint.lite_chat.IM.huanxin.HXResolveConversation
import com.rightpoint.lite_chat.MainActivity
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

/**
 * Description：
 * @author Wonder Wei
 * Create date：2020/7/23 4:00 PM
 */
object ChannelConversation {
    const val CHANNEL_CONVERSATION = "com.rightpoint.litechat/conversation"

    @JvmStatic
    fun connect(flutterEngine: FlutterEngine, activity: MainActivity) {
        val methodChannel = MethodChannel(flutterEngine.dartExecutor,
                CHANNEL_CONVERSATION)
        val iResolveConversation: IResolveConversation = HXResolveConversation()
        methodChannel.setMethodCallHandler { call, result ->
            when (call.method) {
                "getConversations" -> {
                    val conversations = iResolveConversation.conversation
                    activity.runOnUiThread { result.success(conversations) }
                }
                "getMsgList" -> {
                    val username = call.argument<String>("username")
                    val msgList = iResolveConversation.getMsgList(username)
                    activity.runOnUiThread { result.success(msgList) }
                }
                "markAllMessagesAsRead" -> {
                    val username = call.argument<String>("username")
                    val conversation = EMClient.getInstance().chatManager().getConversation(username)
                    conversation?.markAllMessagesAsRead()
                    activity.runOnUiThread { result.success(null) }
                }
            }
        }
    }
}